sphinxgallery/gen_rst.py

```diff
@@ -445,7 +445,7 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf, plot_gallery):
     Returns the set of functions/classes imported in the example.
     """
     base_image_name = os.path.splitext(fname)[0]
-    image_fname = '%s_%%03d.png' % base_image_name
+    image_fname = 'sphx_glr_%s_%%03d.png' % base_image_name
```

Comment:
The sphx-glr prefix is because scripts that don't plot don't start with plot and it is also given to the thumbnails.

Indeed the extra prefix would break the manual calls one does on the documentation of nilearn.

Another thought is if this whole hack of cleaning the images has to be done in the gallery scripts. I see nilearn and scikit-learn(https://github.com/scikit-learn/scikit-learn/blob/master/doc/Makefile#L40) clean the _build/html/_images with an instruction from the Makefile every time one uses `make html`, which seems more robust as it is done before sphinx starts doing anything. I wonder if it then would be best to just remove this hack from the gallery script and write in the Sphinx-gallery documentation to users to insert the extra instruction in their makefile.